backend/server.js

```diff
@@ -368,16 +371,81 @@ app.post('/webhook/', (req, res) => {
 
       if (text === 'test') {
         notifyer.sendFBNotification(sender, 'CS 1800 now has 1 seat available!! Check it out on https://searchneu.com/cs1800 !');
+      } else if (text === 'What is my facebook messenger sender id?') {
+        notifyer.sendFBNotification(sender, sender);
       } else {
         // notifyer.sendFBNotification(sender, "Yo! 👋😃😆 I'm the Search NEU bot. Someday, I will notify you when seats open up in classes that are full. 😎👌🍩 But that day is not today...");
         notifyer.sendFBNotification(sender, "Yo! 👋😃😆 I'm the Search NEU bot. I will notify you when seats open up in classes that are full. Sign up on https://searchneu.com!");
       }
+    } else if (event.optin) {
+      macros.log('Got opt in button click!', event, event.optin.ref);
+
+      // The frontned send a classHash to follow and a list of sectionHashes to follow.
+      let userObject = {};
+      try {
+        userObject = JSON.parse(event.optin.ref);
+      } catch (e) {
+        macros.error('Unable to parse user data from frontend?', event.optin.ref);
+        res.sendStatus(200);
```

Comment:
Shouldn't we be sending `500 Internal Server Error` or `400 Bad Request` instead of `200 OK`? Or do we need to always return 200?

Reply:
From the Facebook docs:

Required 200 OK Response
When you receive a webhook event, you must always return a 200 OK HTTP response. Failing to do so may cause your webhook to be unsubscribed by the Messenger Platform.

https://developers.facebook.com/docs/messenger-platform/webhook

I'll add a comment about this!